File: sample/src/main/java/com/github/abdularis/sample/MainActivity.java
package com.github.abdularis.sample;

import android.os.Handler;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

import com.github.abdularis.buttonprogress.DownloadButtonProgress;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        setListenerFor(R.id.button_progress_1, R.id.subtitle_1);
        setListenerFor(R.id.button_progress_2, R.id.subtitle_2);

        final DownloadButtonProgress btn = findViewById(R.id.button_progress_1);
        btn.addOnStateChangedListeners(new DownloadButtonProgress.OnStateChangedListener() {
            @Override
            public void onStateChanged(int newState) {
                Log.d("TestMe", "state = " + newState);
                if (newState == DownloadButtonProgress.STATE_FINISHED) {
                    btn.setVisibility(View.GONE);
                }
            }
        });
    }

    private void setListenerFor(int btnId, int subtitleId) {
        final DownloadButtonProgress btn = findViewById(btnId);
        final TextView subtitle = findViewById(subtitleId);

        btn.addOnClickListener(new DownloadButtonProgress.OnClickListener() {
            @Override
            public void onIdleButtonClick(View view) {
                new Thread(new SampleTask(new Handler(), btn, subtitle)).start();
            }

            @Override
            public void onCancelButtonClick(View view) {
                // called when cancel button/icon is clicked
            }

            @Override
            public void onFinishButtonClick(View view) {
                // called when finish button/icon is clicked
            }
        });
    }

}

File: sample/src/main/java/com/github/abdularis/sample/MainActivity.kt
package com.github.abdularis.sample

import android.os.Bundle
import android.os.Handler
import android.util.Log
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.github.abdularis.buttonprogress.DownloadButtonProgress
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setListenerFor(button_progress_1)
        setListenerFor(button_progress_2)

        button_progress_1.addOnStateChangedListeners { newState ->
            if (newState == DownloadButtonProgress.STATE_FINISHED) {
                button_progress_1.visibility = View.GONE
            }

            subtitle_1.text = when (newState) {
                DownloadButtonProgress.STATE_IDLE -> "Idle"
                DownloadButtonProgress.STATE_INDETERMINATE -> "Indeterminate"
                DownloadButtonProgress.STATE_DETERMINATE -> "Determinate"
                DownloadButtonProgress.STATE_FINISHED -> "Finished"
                else -> ""
            }
        }

        button_progress_2.addOnStateChangedListeners { newState ->
            subtitle_2.text = when (newState) {
                DownloadButtonProgress.STATE_IDLE -> "Idle"
                DownloadButtonProgress.STATE_INDETERMINATE -> "Indeterminate"
                DownloadButtonProgress.STATE_DETERMINATE -> "Determinate"
                DownloadButtonProgress.STATE_FINISHED -> "Finished"
                else -> ""
            }
        }
    }

    private fun setListenerFor(downloadButtonProgress: DownloadButtonProgress) {
        downloadButtonProgress.addOnClickListener(object : DownloadButtonProgress.OnClickListener {
            override fun onIdleButtonClick(view: View) {
                Thread(SampleTask(Handler(), downloadButtonProgress)).start()
            }

            override fun onCancelButtonClick(view: View) {
                // called when cancel button/icon is clicked
            }

            override fun onFinishButtonClick(view: View) {
                // called when finish button/icon is clicked
            }
        })
    }
}